metrics-operator/controllers/common/analysis/objective_evaluator.go

```diff
@@ -0,0 +1,64 @@
+package analysis
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+)
+
+type ObjectiveEvaluator struct {
+	TargetEvaluator ITargetEvaluator
+}
+
+func NewObjectiveEvaluator(t ITargetEvaluator) ObjectiveEvaluator {
+	return ObjectiveEvaluator{
+		TargetEvaluator: t,
+	}
+}
+
+func (oe *ObjectiveEvaluator) Evaluate(values map[string]string, obj v1alpha3.Objective) types.ObjectiveResult {
+	result := types.ObjectiveResult{
+		Score: 0.0,
+		Value: 0.0,
+	}
+
+	// get the value
+	floatVal, err := getValueFromMap(values, obj.AnalysisValueTemplateRef.Name)
```

Comment:
so the key for the value map is the valueTemplate name, but what happens if I have multiple objectives with the same template ref?

Reply:
would it make sense to introduce an Objective.name field instead?

Reply:
This name should be the name of `AnalysisValueTemplate` CR, so this cannot happen. What actually can happen is that we will have 2 `AnalysisValueTemplate` CRs in two separate namespaces, this will be a problem, I will adapt this part, thank you! :)

Reply:
Can I still have 2 objectives referring to the exact same CR with different thresholds? Eg. pass criteria and warn criteria?

Reply:
yes, you will still have the same value, so that won't be a problem